pkg/object/object.go

```diff
@@ -227,9 +284,82 @@ func (v *Volume) validate() error {
 	return nil
 }
 
+func (sd *SecretData) validate() error {
+	var count int
+	if sd.Plaintext != nil {
+		count++
+	}
+	if sd.Base64 != nil {
+		count++
+	}
+	if sd.File != nil {
+		count++
+	}
+
+	switch count {
```

Comment:
why `switch` here? Wouldn't be better to use `if`?

If you really want `switch` that there should be  `default`  case to make ti future proof.